### DISCUSSION TITLE
Missing total PV power sensor?

### DISCUSSION THREAD

**Discussion by andrewmooreio** [nathanmarlor/foxess_modbus#128] [General]
Hi. I've just switched from the StealthChestnut integration and have noticed there is no total power sensor for PV output. e.g. current total solar generation PV1 + PV2 in W.

I've created a template sensor for now ```{{ (states('sensor.pv1_power') | float(default=0) + states('sensor.pv2_power') | float(default=0) ) * 1 }}```, just wondering if it's possible to hard code it into the integration, as it's a really useful metric that I'm sure a lot of users would appreciate.

There may be other sensors that are provided by templates [here](https://github.com/StealthChesnut/HA-FoxESS-Modbus/blob/main/custom_components/HA-FoxESS-Modbus/configuration.yaml#L85), but the total PV power is a particularly useful one.

Cheers.

**Comment by canton7** [comment]
Fair. There's rpower (Inverter Power), but that includes battery output.

**Reply by canton7** [reply]
Created #129 to track

**Reply by screenagerbe** [reply]
AFAIK those sensors are available in v1.4.0b7:
![image](https://user-images.githubusercontent.com/82099983/230668557-02c42553-7589-42e4-91ab-4a9bf9d25c81.png)

But, there are no sensors for PV1 Power Daily and PV2 Power Daily where there are PV1/2 Power Total sensors available.
I rather have them separate as PV1 and PV2 have different orientations.

I know I could create the missing sensors manually, but seems a bit strange to create just this two sensors when there are other Daily sensors available in v1.0.4.b7

**Reply by canton7** [reply]
That's different. OP is asking for PV1 + PV2. You're referring to the sum of PV1 over time.

I don't think we should be creating every sensor type which someone could conceivably want, to be honest. You can create a utility meter from PV1 Total to show daily usage in a couple of clicks. We put in PV1 total as the energy dashboard needs it, so we needed it in order to automatically configure the energy dashboard.

**Reply by screenagerbe** [reply]
ok, makes sense :)
I'll handle it the same way as offpeak and peak tariffs
thanks.

**Reply by canton7** [reply]
(or if we do, maybe make them opt-in, either through a settings page or by setting them to disabled initially?)

**Reply by canton7** [reply]
I think "Solar Generation Total" / "Solar Generation Today" cover this?

**Reply by canton7** [reply]
Ah no, those aren't in W.
